File: LaboratoryWork/Semester_3/Laboratory_2/GoLang/cmd/main/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"os/exec"
	"strconv"
	"strings"

	"laba2/tasks"
	"laba2/utils/menu"
)

type MenuOptions int

const (
	Exit MenuOptions = iota
	ArithmeticExpression
	InteractionWithSet
	SetPartition
	Subarrays
	CheckBalance
	ConvolutionMethod
	LRUCache
)

func clearScreen() {
	cmd := exec.Command("clear")
	cmd.Stdout = os.Stdout
	_ = cmd.Run()
}

func main() {
	clearScreen()
	defer func() {
		if r := recover(); r != nil {
			fmt.Println("Ошибка:", r)
		}
	}()

	args := os.Args
	choice := -1

	if len(args) == 5 {
		choice = int(InteractionWithSet)
	} else {
		menu.StartMenu()
		reader := bufio.NewReader(os.Stdin)
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		val, err := strconv.Atoi(input)
		if err != nil {
			fmt.Println("Неверный ввод, попробуйте снова.")
			return
		}
		choice = val
	}

	switch MenuOptions(choice) {
		case Exit:
			clearScreen()
			return

		case ArithmeticExpression:
			tasks.ArithmeticExpression()

		case InteractionWithSet:
			err := tasks.InteractionWithSet(args)
			if err != nil {
				fmt.Println("Ошибка:", err)
			}

		case SetPartition:
			tasks.SetPartition()

		case Subarrays:
			tasks.Subarrays()

		case CheckBalance:
			tasks.AVLBalance()

		case ConvolutionMethod:
			tasks.IntWithHashTables()

		case LRUCache:
			tasks.InteractionWithLRU()

		default:
			clearScreen()
			fmt.Println("Неверный выбор. Попробуйте снова.")
	}
}

File: LaboratoryWork/Semester_3/Laboratory_2/C++/main.cpp
#include <iostream>
#include <fstream>

#include "include/menu.hpp"
#include "include/ArithmeticExpression.hpp"
#include "include/SetPartition.hpp"
#include "include/InteractionWithDoubleHashSet.hpp"
#include "include/Subarrays.hpp"
#include "include/AVLBalance.hpp"
#include "include/InteractionWithHashs.hpp"
#include "include/InteractionWithLRUHash.hpp"

using namespace std;

int main(int argc, char* argv[]) {
    system("clear");
    try {
        int choice = -1;
        if (argc == 5) {
            choice = 2;
        } else {
            startMenu();
            cin >> choice;
        }

        switch (static_cast<MenuOptions>(choice)) {
            case MenuOptions::Exit:
                system("clear");
                return 0;
            case MenuOptions::ArithmeticExpression:
                cin.clear();
                cin.ignore();
                arithmeticExpression();
                break;
            case MenuOptions::InteractionWithSet:
                interactionWithSet(argc, argv);
                break;
            case MenuOptions::SetPartition:
                setPartition();
                break;
            case MenuOptions::Subarrays:
                subarrays();
                break;
            case MenuOptions::CheckBalance:
                AVLBalance();
                break;
            case MenuOptions::ConvolutionMethod:
                intWithHashs();
                break;
            case MenuOptions::LRUCache:
                interactionWithLRU();
                break;
            default:
                system("clear");
                cout << "Неверный выбор. Попробуйте снова" << endl;
                break;
        }
    } catch (exception& e) {
        cout << e.what() << endl;
    }
}